### PR TITLE
Almost Always Avoid Auto in i_at

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
Noticed a bunch of egregious uses of auto while doing the map reference thing.

#### Describe the solution
Un-auto the statements.

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -9693,8 +9693,8 @@ void Character::migrate_items_to_storage( bool disintegrate )
 std::string Character::is_snuggling() const
 {
     map &here = get_map();
-    auto begin = here.i_at( pos() ).begin();
-    auto end = here.i_at( pos() ).end();
+    map_stack begin = here.i_at( pos() ).begin();
+    map_stack end = here.i_at( pos() ).end();
```
Comment (kevingranade) [inline src/character.cpp:9697]:
```suggestion
    auto begin = here.i_at( pos() ).begin();
    auto end = here.i_at( pos() ).end();
```

File: src/computer_session.cpp [hunk 2]
```diff
@@ -980,7 +980,7 @@ void computer_session::action_irradiator()
                 print_error( _( "ERROR: Processing platform empty." ) );
             } else {
                 g->u.moves -= 300;
-                for( auto it = here.i_at( dest ).begin(); it != here.i_at( dest ).end(); ++it ) {
+                for( item it = here.i_at( dest ).begin(); it != here.i_at( dest ).end(); ++it ) {
```
Comment (ZhilkinSerg) [inline src/computer_session.cpp:983]:
```suggestion
                for( auto it = here.i_at( dest ).begin(); it != here.i_at( dest ).end(); ++it ) {
```

File: src/game.cpp [hunk 3]
```diff
@@ -10144,7 +10144,7 @@ bool game::grabbed_furn_move( const tripoint &dp )
             std::move( m.i_at( fpos ).begin(), m.i_at( fpos ).end(),
                        std::back_inserter( temp ) );
             m.i_clear( fpos );
-            for( auto item_iter = m.i_at( fdest ).begin();
+            for( item item_iter = m.i_at( fdest ).begin();
```
Comment (ZhilkinSerg) [inline src/game.cpp:10147]:
```suggestion
            for( auto item_iter = m.i_at( fdest ).begin();
```

**Comment by meladath** [comment]
So I love auto, but damn, why would anyone use auto for a typename thats 4 letters long? Boggles my mind.

**Pull request merged**